### PR TITLE
std.algorithm.iteration: Fix a -dip1000 compilable issue

### PR DESCRIPTION
**Pull request by carblue**
https://github.com/dlang/phobos/blob/master/dip1000.mak with
aa[std.algorithm.iteration]=-dip1000
Errors when running: make -f posix.mak std/algorithm/iteration.test
...
```
T=`mktemp -d /tmp/.dmd-run-test.XXXXXX` &&                                                              \
  (                                                                                                     \
    ../dmd/generated/linux/release/64/dmd -od$T -conf= -I../druntime/import  -w -de -dip25 -m64 -fPIC -transition=complex -O -release -dip1000  -main -unittest generated/linux/release/64/libphobos2.a -defaultlib= -debuglib= -L-ldl -cov -run std/algorithm/iteration.d ;     \
    RET=$? ; rm -rf $T ; exit $RET                                                                   \
  )
std/algorithm/iteration.d(4245): Error: @safe function std.algorithm.iteration.__unittest_L4227_C7 cannot call @system function std.algorithm.comparison.equal!().equal!(Result, string[]).equal
std/algorithm/iteration.d(5525): Error: static assert:  is(typeof(sum(R)(R r) if (isInputRange!R && !isInfinite!R && is(typeof(r.front + r.front)))((__error)[])) == double) is false
```
The remaining errors are from std/container/slist.d only

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -9806,7 +9806,7 @@ if (isInputRange!Range)
     {
         return predFun(rhs, lhs);
     }
-    private Range _input;
+    /*private*/ Range _input;
```
Comment (wilzbach) [inline std/range/package.d]: That's not really a proper fix and we shouldn't revert to such hacks :/

Comment (WalterBright) [reply std/range/package.d]: Why does making it public make it safe?

Comment (wilzbach) [reply std/range/package.d]: I just tried this locally and if I set `@trusted:` to `std.container.slist`, I can compile + test std.algorithm.iteration successfully with -dip1000 (e.g. `pmake std/algorithm/iteration.test`)
-> So I suggest to drop this specific change.

Comment (carblue) [reply std/range/package.d]: @wilzbach Done as suggested

**Pull request merged**